examples/azure.ts

```diff
@@ -1,14 +1,19 @@
 #!/usr/bin/env -S npm run tsn -T
 
 import { AzureOpenAI } from 'openai';
+import { getBearerTokenProvider, DefaultAzureCredential } from '@azure/identity';
 
 // Corresponds to your Model deployment within your OpenAI resource, e.g. gpt-4-1106-preview
 // Navigate to the Azure OpenAI Studio to deploy a model.
 const deployment = 'gpt-4-1106-preview';
 
+const credential = new DefaultAzureCredential();
+const scope = 'https://cognitiveservices.azure.com/.default';
+const azureADTokenProvider = getBearerTokenProvider(credential, scope);
```

Comment:
question: is this the most common authentication pattern for Azure OpenAI users?

Reply:
Yes, it is Microsoft's enterprise authentication solution.